backend/metadata_writer/src/metadata_writer.py

```diff
@@ -83,11 +84,21 @@ def output_name_to_argo(name: str) -> str:
     import re
     return re.sub('-+', '-', re.sub('[^-0-9a-z]+', '-', name.lower())).strip('-')
 
+def is_s3_endpoint(endpoint: str) -> bool:
+    return re.search('^.*s3.*amazonaws.com.*$', endpoint)
+
+def get_object_store_provider(endpoint: str) -> bool:
+    if is_s3_endpoint(endpoint):
+        return 's3'
+    else:
+        return 'minio'
 
 def argo_artifact_to_uri(artifact: dict) -> str:
+    # s3 here means s3 compatible object storage. not AWS S3.
     if 's3' in artifact:
         s3_artifact = artifact['s3']
-        return 'minio://{bucket}/{key}'.format(
+        return '{provider}://{bucket}/{key}'.format(
```

Comment:
What is we always use `s3://<endpoint>/bucket/key` format in both Metadata Writer and the UX? (or 's3_with_endpoint://...').
This way we can retain all data in the single URI for Minio, GCS and AWS.

Reply:
Let's have the discussion in #3531